app/src/main/java/com/github/se/eventradar/ui/qrCode/QrCodeTicketUi.kt

```diff
@@ -0,0 +1,280 @@
+        ScanTicketQrViewModel.Action.FirebaseUpdateError,
+        ScanTicketQrViewModel.Action.FirebaseFetchError,
+        ScanTicketQrViewModel.Action.AnalyserError -> {
+          EntryDialog(2, viewModel)
+        }
+      }
+    }
+    BottomNavigationMenu(
+        onTabSelected = { tab ->
+          navigationActions.navigateTo(tab)
+          viewModel.resetConditions()
+        },
+        tabList = TOP_LEVEL_DESTINATIONS,
+        selectedItem = TOP_LEVEL_DESTINATIONS[3],
+        modifier =
+            Modifier.testTag("bottomNavMenu").constrainAs(bottomNav) {
+              bottom.linkTo(parent.bottom)
+              start.linkTo(parent.start)
+              end.linkTo(parent.end)
+            })
+  }
+}
+
+@Composable
+fun EntryDialog(edr: Int, viewModel: ScanTicketQrViewModel) {
+  Dialog(onDismissRequest = { viewModel.changeAction(ScanTicketQrViewModel.Action.ScanTicket) }) {
+    val boxColor =
+        when (edr) {
+          0 -> Color.Green
+          1 -> Color.Red
+          else -> Color.Yellow
+        }
+
+    Box(
+        modifier =
+            Modifier.size(400.dp) // Adjust the size here to make it larger
+                .background(boxColor, RoundedCornerShape(8.dp))
+                .padding(20.dp)
+                .semantics {
+                  testTag =
+                      when (boxColor) {
+                        Color.Green -> "ApprovedBox"
+                        Color.Red -> "DeniedBox"
+                        else -> "ErrorBox"
+                      }
+                },
+        // Aligning content to the top right corner
+    ) {
+      Column(
+          modifier = Modifier.fillMaxSize().padding(vertical = 16.dp),
+          verticalArrangement = Arrangement.Center,
+          horizontalAlignment = Alignment.CenterHorizontally) {
+            val textToShow =
+                when (edr) {
+                  0 -> "Entry Approved"
+                  1 -> "Entry Denied"
+                  else -> "Error, Please Retry"
+                }
+            Text(
+                text = textToShow,
+                style = TextStyle(fontWeight = FontWeight.Bold, fontSize = 24.sp),
+                modifier =
+                    Modifier.semantics {
+                      testTag =
+                          when (edr) {
+                            0 -> "EntryApprovedText"
+                            1 -> "EntryDeniedText"
+                            else -> "ErrorText"
+                          }
+                    })
+          }
+      Box(modifier = Modifier.align(Alignment.TopEnd).padding(8.dp)) {
+        IconButton(
+            onClick = { viewModel.changeAction(ScanTicketQrViewModel.Action.ScanTicket) },
+            modifier =
+                Modifier.semantics { testTag = "closeButton" } // Adding testTag to the IconButton
+            ) {
+              Icon(Icons.Default.Close, contentDescription = "Close")
+            }
+      }
+    }
+  }
+}
+
+@Preview(showBackground = true)
```

Comment:
Here you can add the tag @ExcludeFromJacocoGeneratedReport to not include this non-functional code in the coverage calculations.